frontend/endpoints/destinations.go

```diff
@@ -101,3 +118,141 @@ func GetDestinationDetails(c *gin.Context) {
 		"error": "destination not found",
 	})
 }
+
+func GetDestinations(c *gin.Context, odigosna string) {
+	dests, err := kube.DefaultClient.OdigosClient.Destinations(odigosna).List(c, metav1.ListOptions{})
+	if err != nil {
+		returnError(c, err)
+		return
+	}
+
+	var resp []Destination
+	for _, dest := range dests.Items {
+		endpointDest := k8sDestinationToEndpointFormat(dest)
+		resp = append(resp, endpointDest)
+	}
+
+	c.JSON(200, resp)
+}
+
+func GetDestinationByName(c *gin.Context, odigosns string) {
+	destName := c.Param("name")
+	destination, err := kube.DefaultClient.OdigosClient.Destinations(odigosns).Get(c, destName, metav1.GetOptions{})
+	if err != nil {
+		returnError(c, err)
+		return
+	}
+
+	resp := k8sDestinationToEndpointFormat(*destination)
+	c.JSON(200, resp)
+}
+
+func CreateNewDestination(c *gin.Context, odigosns string) {
+
+	request := Destination{}
+	if err := c.ShouldBindJSON(&request); err != nil {
+		returnError(c, err)
+		return
+	}
+
+	k8sDestination := v1alpha1.Destination{
+		TypeMeta:   metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: request.Name,
+		},
+		Spec:       v1alpha1.DestinationSpec{
+			Type:      request.Type,
+			Data:      request.Data,
```

Comment:
Sensitive fields should be persisted as Kubernetes secret and linked via `secretRef` property